test/clj/rems/api/test_applications.clj

```diff
@@ -1454,6 +1453,112 @@
                  (first (:application/members application))
                  (get-in application [:application/events 2 :application/member]))))))))
 
+
+(deftest test-application-vote
+  (let [api-key "42"
+        applicant-id "alice"
+        handler-id1 "handler"
+        handler-id2 "developer"
+        reviewer-id "carl"
+        owner "owner"
+        form-id (test-helpers/create-form! {})
+        workflow-id (test-helpers/create-workflow! {:type :workflow/master
+                                                    :handlers [handler-id1 handler-id2]})
+        cat-item-id (test-helpers/create-catalogue-item! {:workflow-id workflow-id
+                                                          :form-id form-id})
+        application-id (test-helpers/create-application! {:catalogue-item-ids [cat-item-id]
+                                                          :actor applicant-id})]
+    (testing "submit"
+      (is (= {:success true} (send-command applicant-id
+                                           {:type :application.command/submit
+                                            :application-id application-id}))))
+    (testing "invite reviewer"
+      (is (= {:success true} (send-command handler-id1
+                                           {:type :application.command/request-review
+                                            :application-id application-id
+                                            :reviewers [reviewer-id]}))))
+    (testing "unrelated user can't vote"
+      (is (= {:success false
+              :errors [{:type "forbidden"}]}
+             (send-command applicant-id
+                           {:type :application.command/vote
+                            :application-id application-id
+                            :vote "Accept"})))
+
+      (is (= {:success false
+              :errors [{:type "forbidden"}]}
+             (send-command reviewer-id
+                           {:type :application.command/vote
+                            :application-id application-id
+                            :vote "Accept"}))))
+
+    (testing "handler can't vote if voting is not enabled"
+      (is (= {:success false
+              :errors [{:type "forbidden"}]}
+             (send-command handler-id1
+                           {:type :application.command/vote
+                            :application-id application-id
+                            :vote "Accept"}))))
+
+    (testing "enable voting"
+      (-> (request :put "/api/workflows/edit")
+          (authenticate api-key owner)
+          (json-body {:id workflow-id
+                      :organization {:organization/id "abc"}
+                      :title "Workflow with voting"
+                      :handlers [handler-id1 handler-id2]
+                      :disable-commands []
+                      :voting {:type "handlers-vote"}})
+          handler
+          read-ok-body
+          :id))
```

Comment:
test should be wrapped inside `(is)` (otherwise test fail won't register properly?)

Reply:
There was an assert in `read-ok-body` but it makes sense to check return value too.